wgengine/netstack/netstack.go

```diff
@@ -1202,16 +1191,14 @@ func stringifyTEI(tei stack.TransportEndpointID) string {
 }
 
 func ipPortOfNetstackAddr(a tcpip.Address, port uint16) (ipp netip.AddrPort, ok bool) {
-	var a16 [16]byte
-	copy(a16[:], a)
-	switch len(a) {
+	switch a.Len() {
 	case 4:
 		return netip.AddrPortFrom(
-			netip.AddrFrom4(*(*[4]byte)(a16[:4])).Unmap(),
+			netip.AddrFrom4(a.As4()).Unmap(),
 			port,
 		), true
 	case 16:
-		return netip.AddrPortFrom(netip.AddrFrom16(a16).Unmap(), port), true
+		return netip.AddrPortFrom(netip.AddrFrom16(a.As16()).Unmap(), port), true
```

Comment:
Does `tcpip.Address.As16` not panic and do the wrapping thing for IPv4 addresses?

If so, you can just use only this line and ditch the switch on Len?

Reply:
Yes, in fact there's a much shorter version, though it drops unmap. I have mixed feelings about that, netstack is _specific_ about mapping, so I believe this to be correct, but I'm wary of it being not directly related. It's not a bad time to land those changes though.